client/app/states/services/custom_button_details/custom_button_details.state.js

```diff
@@ -63,7 +63,21 @@ function StateController ($state, $stateParams, CollectionsApi, EventNotificatio
   }
   init()
   function refreshField (field) {
-    return DialogFieldRefresh.refreshDialogField(vm.dialogData, [field.name], vm.dialogUrl, vm.dialogId)
+    let targetType = 'service'
+    let targetId = vm.serviceId
+    if (vm.vmId) {
+      targetType = 'vm'
+      targetId = vm.vmId
+    }
+
+    let idList = {
+      dialogId: vm.dialogId,
+      resourceActionId: $stateParams.button.resource_action.id,
```

Comment:
This is more of a question than anything else.  Are we assured this particular state param will always be there?  If so, the I am fine with it , I think the other thing to consider is more of a style thing.  I tend to like to put things into variables that can be reused.  I don't know if we have a use for $stateParams.button more than just this statement but if we do, we should do a 
const button = $stateParams.button

resourceActionId: button.resource_action.id,

Reply:
@eclarizio ☝️ 🤔

Reply:
`$stateParams.button` should always be there, yes, since it is a reference to the custom button that was just clicked. Likewise, `resource_action` on that custom button should also always be there.

I'll add `vm.resourceAction` since we already have a `vm.button`.